hudi-client/hudi-client-common/src/main/java/org/apache/hudi/client/transaction/lock/metrics/HoodieLockMetrics.java

```diff
@@ -70,11 +71,13 @@ private String getMetricsName(String metric) {
 
   private Timer createTimerForMetrics(MetricRegistry registry, String metric) {
     String metricName = getMetricsName(metric);
-    synchronized (REGISTRY_LOCK) {
-      if (registry.getMetrics().get(metricName) == null) {
-        lockDuration = new Timer(new SlidingWindowReservoir(keepLastNtimes));
-        registry.register(metricName, lockDuration);
-        return lockDuration;
+    if (registry.getMetrics().get(metricName) == null) {
+      synchronized (Registry.class) {
+        if (registry.getMetrics().get(metricName) == null) {
```

Comment:
Why the `REGISTRY_LOCK` can not work correctly here, can you elaborate the explanation then ?

Reply:
> 为什么`REGISTRY_LOCK`不能在这里正常工作，你能详细解释一下吗？

I have updated the description

Reply:
I mean the `REGISTRY_LOCK` can still be used instead of the class obj lock `Registry`.

Reply:
Yes, it is not necessary to lock the whole class. I will make the correction in these two days